src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,222 @@
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt.prefSettingAdjuster");
+
+
+/*
+ * The following transform functions, valueInputToBinder and binderToValueInput
+ * sync the value being edited from our model to the adjust controler, and in
+ * this situation take in to account the semantics for binding boolean values by
+ * looking at the metadata schema of the term currently being edited, and if it
+ * is boolean applying that semantic.
+ *
+ * At the moment these are looking up the prefSettingAdjuster component manually,
+ * and should be changed to the correct way to set those values, and also prepare
+ * for a future where there is more than one instance of a pref adjuster at a time
+ * on a page.  They should really just be subcomponents on the prefSettingAdjuster
+ * component, and be able to look at the item currently being edited to apply the
+ * correct semantics. (Which could be even more complex in the future, at the moment
+ * we only handle very simple types.)
+ */
+
+fluid.defaults("gpii.devpmt.prefSettingAdjuster.valueInputToBinder", {
+    gradeNames: ["fluid.standardTransformFunction"]
+});
+
+gpii.devpmt.prefSettingAdjuster.valueInputToBinder = function (value) {
+    var adjuster = fluid.queryIoCSelector(fluid.rootComponent, "gpii.devpmt.prefSettingAdjuster")[0];
```

Comment:
This piece of schema needs to be injected into the component on startup and referenced in the transformation document and appear here in arguments, rather than via this horrifically global side-effect on the IoC resolution system. The intention is that all model transformation functions are pure functions.

Reply:
A future version of the language that we will use to replace JavaScript will prohibit the use of arbitrary global access in functions from these contexts as "bad" : P https://wiki.fluidproject.org/display/fluid/A+Good+Function

Reply:
this schema comes in as an inputVariable to the transform now